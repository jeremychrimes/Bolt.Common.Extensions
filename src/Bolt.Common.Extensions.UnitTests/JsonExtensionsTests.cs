using System;
using Bolt.Common.Extensions.UnitTests.Infra;
using Shouldly;
using Xunit;

namespace Bolt.Common.Extensions.UnitTests
{
    public class JsonExtensionsTests
    {
        [Fact]
        public void ToJsonTest()
        {
            var person = new Person
            {
                Name = "test-name",
                Age = 58,
                CreatedAt = new DateTime(2021, 1, 1, 0, 0, 0, DateTimeKind.Utc)
            };

            var serializedData = person.SerializeToJson();

            serializedData.ShouldMatchApprovedDefault();
        }

        [Fact]
        public void ToPrettyJsonTest()
        {
            var person = new Person
            {
                Name = "test-name",
                Age = 58,
                CreatedAt = new DateTime(2021, 1, 1, 0, 0, 0, DateTimeKind.Utc)
            };

            var serializedData = person.SerializeToPrettyJson();

            serializedData.ShouldMatchApprovedDefault();
        }

        [Fact]
        public void FromJsonTest()
        {
            var person = new Person
            {
                Name = "test-name",
                Age = 58,
                CreatedAt = new DateTime(2021, 1, 1, 0, 0, 0, DateTimeKind.Utc)
            };

            var serializedData = person.SerializeToJson();

            var result = serializedData.DeserializeFromJson<Person>();

            result.ShouldSatisfyAllConditions
            (
                () => result.Name.ShouldBe(person.Name),
                () => result.Age.ShouldBe(person.Age),
                () => result.CreatedAt.ShouldBe(person.CreatedAt)
            );
        }
    }

    public record Person
    {
        public string Name { get; init; }
        public int Age { get; init; }
        public DateTime CreatedAt { get; init; }
    }
}
