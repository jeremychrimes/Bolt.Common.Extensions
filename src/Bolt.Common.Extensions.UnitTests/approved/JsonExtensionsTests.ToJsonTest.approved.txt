{"name":"test-name","age":58,"createdAt":"2021-01-01T00:00:00Z"}